core/src/main/java/org/apache/accumulo/core/client/mapred/AbstractInputFormat.java

```diff
@@ -384,7 +387,21 @@ public static InputTableConfig getInputTableConfig(JobConf job, String tableName
   protected abstract static class AbstractRecordReader<K,V> implements RecordReader<K,V> {
     protected long numKeysRead;
     protected Iterator<Map.Entry<Key,Value>> scannerIterator;
-    protected RangeInputSplit split;
+    protected org.apache.accumulo.core.client.mapreduce.impl.AccumuloInputSplit split;
+    protected ScannerBase scannerBase;
+
+    /**
+     * Configures the iterators on a scanner for the given table name.
+     *
+     * @param job
+     *          the Hadoop job configuration
+     * @param scanner
+     *          the scanner for which to configure the iterators
+     * @param tableName
+     *          the table name for which the scanner is configured
+     * @since 1.7.0
+     */
+    protected abstract void setupIterators(JobConf job, ScannerBase scanner, String tableName, AccumuloInputSplit split);
```

Comment:
That's a good point. In an effort to avoid mistakes of the past and keep AccumulInputSplit out of public API how about something like this:

```java
// InputFormatBase
List<IteratorSetting> contextIterators(TaskAttemptContext context, String tableName) {
  return getIterators(context).getIterators();
}

// AccumululoMultiTableInputFormat
List<IteratorSetting> contextIterators(TaskAttemptContext context, String tableName) {
  return getInputTableConfig(context, tableName).getIterators();
}

// lift setupIterators to AbstractInputFormat (no longer abstract)
protected abstract List<IteratorSetting> contextIterators(TaskAttemptContext context, String tableName);

private void setupIterators(TaskAttemptContext context, ScannerBase scanner, String tableName, AccumuloInputSplit split) {
    List<IteratorSetting> iterators = null;
    if (null == split) {
      iterators = contextIterators(context);
    } else {
      iterators = split.getIterators();
      if (null == iterators) {
        iterators = contextIterators(context);
      }
    }
    for (IteratorSetting iterator : iterators)
      scanner.addScanIterator(iterator);
}
```
Looking at code right now the only part that diverges is pulling out default split if they're somehow unavailable in the split. This narrows the interface, hopefully in an acceptable way?